plugin/storage/es/spanstore/writer.go

```diff
@@ -98,21 +107,12 @@ func (s *SpanWriter) CreateTemplates(spanTemplate, serviceTemplate string, index
 // spanAndServiceIndexFn returns names of span and service indices
 type spanAndServiceIndexFn func(spanTime time.Time) (string, string)
 
-func getSpanAndServiceIndexFn(p SpanWriterParams) spanAndServiceIndexFn {
+func getSpanAndServiceIndexFn(p SpanWriterParams, writeAlias string) spanAndServiceIndexFn {
 	spanIndexPrefix := p.IndexPrefix.Apply(spanIndexBaseName)
 	serviceIndexPrefix := p.IndexPrefix.Apply(serviceIndexBaseName)
-	if p.Archive {
+	if writeAlias != "" {
 		return func(_ time.Time) (string, string) {
-			if p.UseReadWriteAliases {
-				return archiveIndex(spanIndexPrefix, archiveWriteIndexSuffix), ""
-			}
-			return archiveIndex(spanIndexPrefix, archiveIndexSuffix), ""
-		}
-	}
-
-	if p.UseReadWriteAliases {
```

Comment:
similar to reader, please continue using this boolean condition instead of `alias!=""`